subtests/docker_cli/systemd/systemd.py

```diff
@@ -65,22 +58,32 @@ def sysd_action(action=None, sysd_unit=None):
             command = '{} {}'.format(command, sysd_unit)
         utils.run(command, ignore_status=False)
 
-    @staticmethod
-    def read_socket(host, port):
-        max_read_tries = 10
-        num_read_try = max_read_tries
-        while num_read_try:
-            # using timeout 13-sec as it's 1-sec longer than two DNS timeouts
-            s = socket.create_connection((host, port), 13)
-            s.settimeout(2.0)
-            try:
-                time_from_socket = s.recv(512).strip()
-                if time_from_socket:
-                    s.shutdown(socket.SHUT_RDWR)
-                    return time_from_socket
-                num_read_try -= 1
-            except socket.timeout:
-                pass
-            time.sleep(1.0)
-        raise DockerTestError("Failed to read from socket with %d tries"
-                              % max_read_tries)
+    def write_unitfile(self, var_to_edit):
+        with open(self.sub_stuff['unit_src'], "r") as srcfile:
+            lines = srcfile.readlines()
+        with open(self.sub_stuff['unit_dest'], "w") as tmpfile:
```

Comment:
`tmpfile` here is dangerously misleading! You can do two things here:

  * Rename variable to `outfile`, or
  * go back to your (correct and proper) earlier way, in which you write a tempfile and then atomically rename it into place.

The second form is correct, because it is super-dangerous to write to non-tempfiles: if your program crashes, or the disk fills up, or there's any of a million different errors, you end up with a corrupt `/etc/systemd/system/foo.unit`.